src/main/scala/sbtassembly/AssemblyUtils.scala

```diff
@@ -68,8 +68,14 @@ object AssemblyUtils {
                 fileOutputStream(false)(target) { out => IO.transfer(from, out) }
               }
             }
-            if(preserveLastModified)
-              target.setLastModified(entry.getTime)
+
+            if (preserveLastModified) {
+              val t: Long = entry.getTime
+              val EPOCH_1980_01_01 = 315532800000L
+              val EPOCH_2010_01_01 = 1262304000000L
+              if (t > EPOCH_1980_01_01) target.setLastModified(t)
```

Comment:
is it better to go with `t >= EPOCH_1980_01_01` ?

Reply:
Actually there are many edge cases around 1980-01-01 like UTC vs local and https://bugs.openjdk.java.net/browse/JDK-8246129, it might just be safer to pick more later dates.